lib/datadog/core/telemetry/metric_worker.rb

```diff
@@ -0,0 +1,35 @@
+# frozen_string_literal: true
+
+require_relative '../worker'
+require_relative '../workers/polling'
+
+module Datadog
+  module Core
+    module Telemetry
+      # Periodically (every DEFAULT_INTERVAL_SECONDS) sends metrics to the telemetry API.
+      class MetricWorker < Core::Worker
+        include Core::Workers::Polling
```

Comment:
So I'm not the one to talk a lot, the profiler already uses 3 threads. But... do we really really want to add another background thread to dd-trace-rb?

If we enable every product, we may be getting close to 9 or 10 now which seems to add up to quite a bit...

Reply:
That is an excellent point.

And since this iteration, we use the same interval time as the heartbeat, we could use the same thread to send the heartbeat and the metrics. 

I will change the code to reflect that. I might have to rename `Heartbeat` to something like `TelemetryWorker` 😄

Reply:
Done https://github.com/DataDog/dd-trace-rb/pull/3230/commits/4826434cdde86dad5e31d3ea1eaeee822e5e3bf5